### PR TITLE
Fix Span names for 404 and when Micrometer is available and disabled

### LINKED ISSUE #26216: OpenTelemetry: By default high HTTP span name cardinality for HTTP server spans
### Describe the bug

This is a follow-up of #16952.
When using the `quarkus-opentelemetry` extension and observing Vert.x HTTP server request spans, the default is still to see span names containing the full request URI path. This means you will possibly get a whole range of span names like
```
/products/myProductId123
/products/myProductId234
/someWeirdPathTryingToExploitSomeVulnerability?someParamX
```
The [OTEL spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#name) declares such high cardinality span names as unsuitable. In practice, this means the span selection combobox in the Jaeger UI becomes unusable.

#17676 was supposed to fix this by introducing span names with parameterized paths (e..g `/products/{productId}`), but there are quite some steps involved to fully prevent the high cardinality span names.

- First of all, the Vert.x HTTP server metrics have to be enabled by setting
`quarkus.micrometer.binder.http-server.enabled=true`.
Without that, no parameterized paths get used.
- Then a "match-all" Vert.x Router route has to be added. That route must either have a name set or `ctx.request().routed(someName)` has to be invoked in its handler. This prevents full request URI span names for any unsupported request paths (that lead to a 404 response).
- The "match-all" route also needs a failure handler, so that invalid HTTP requests don't show up in tracing with their full request paths. (But you want to ensure that the failure handler route name isn't used as span name for all requests where a _specific_ route already got matched - you want the specific route name as span name then (see example code below).)

<details>
<summary>Example matchAll route</summary>

~~~
        final Route matchAllRoute = router.route();
        matchAllRoute.failureHandler(ctx -> {
            if (ctx.get(KEY_MATCH_ALL_ROUTE_APPLIED) == null) {
                // handler of matchAllRoute not applied, meaning this request is invalid/failed from the start;
                // ensure span name is set to fixed string instead of the request path
                ctx.request().routed(MATCH_ALL_ROUTE_NAME);
            }
            ctx.next();
        });
        matchAllRoute.handler(ctx -> {
            // ensure span name is set to fixed string instead of the request path
            ctx.request().routed(MATCH_ALL_ROUTE_NAME);
            ctx.put(KEY_MATCH_ALL_ROUTE_APPLIED, true);
            ctx.next();
        });
~~~

</details>

FMPOV, the issue here is that
- without Vert.x HTTP server metrics enabled, still the full request URI paths get used as span names for all requests
- (with metrics enabled) the default span name is still the full request URI path, when no named route gets applied (or `ctx.request().routed(name)` isn't invoked).

Also there is documentation missing on the need to enable Vert.x HTTP server metrics here.

### Expected behavior

I would expect the HTTP request method name (`GET`, `POST`, `PUT`, etc.) to be used as span name if Vert.x HTTP server metrics are not enabled and also if no request route name got set/applied (i.e. no named route got matched and `ctx.request().routed(name)` wasn't invoked).

It seems to me, this can be implemented by just letting the `route(HttpRequest)` method in `HttpInstrumenterVertxTracer.ServerAttributesExtractor` always return `null`.

### Actual behavior

High cardinality span names are used by default.

### How to Reproduce?

Make HTTP requests to quarkus-opentelemetry application having Vert.x HTTP server endpoint.
Also make requests with unsupported paths, leading to 404 responses. And make invalid HTTP requests (e.g. with `printf 'GET invalid HTTP/1.0\r\n\r\n' | nc -q 3 localhost 80`).
Observe span names in Jaeger UI.

### Output of `uname -a` or `ver`

`Linux be6z00br-vm 5.4.0-117-generic #132-Ubuntu SMP Thu Jun 2 00:39:06 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`

### Output of `java -version`

openjdk 17.0.3 2022-04-19

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.8.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by radcortez**
- Fixes #26216

### CODE REVIEW

**Review by brunobat — APPROVED** [review]
Looks good to me. Thanks @radcortez

File: extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/tracing/TracerProcessor.java [hunk 1]
```diff
@@ -60,7 +61,8 @@ static class MetricsExtensionAvailable implements BooleanSupplier {
 
         @Override
         public boolean getAsBoolean() {
-            return IS_MICROMETER_EXTENSION_AVAILABLE;
+            return IS_MICROMETER_EXTENSION_AVAILABLE && ConfigProvider.getConfig()
+                    .getOptionalValue("quarkus.micrometer.binder.http-server.enabled", Boolean.class).orElse(true);
```
Comment (calohmn) [inline extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/tracing/TracerProcessor.java:65]: @radcortez Found an issue here when testing this:
The default for [quarkus.micrometer.binder.http-server.enabled](https://quarkus.io/guides/micrometer#quarkus-micrometer_quarkus.micrometer.binder.http-server.enabled) is not necessarily `true`.

If `quarkus.micrometer.binder-enabled-default` is `false`, then `quarkus.micrometer.binder.http-server.enabled` is also `false` by default.

Comment (radcortez) [reply extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/tracing/TracerProcessor.java:65]: Ok, I'll check.

Comment (radcortez) [reply extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/tracing/TracerProcessor.java:65]: This should be the proper check:
https://github.com/quarkusio/quarkus/blob/8816386b87e145ade6ef55aa0d7572f20c6eec21/extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/config/MicrometerConfig.java#L70-L80

Comment (radcortez) [reply extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/tracing/TracerProcessor.java:65]: In https://github.com/quarkusio/quarkus/pull/26871.

**Pull request merged**